third_party/move/move-prover/boogie-backend/src/options.rs

```diff
@@ -19,9 +19,22 @@ const DEFAULT_BOOGIE_FLAGS: &[&str] = &[
     "-proverOpt:O:model_validate=true",
 ];
 
-const MIN_BOOGIE_VERSION: &str = "2.15.8";
-const MIN_Z3_VERSION: &str = "4.11.0";
-const MIN_CVC5_VERSION: &str = "0.0.3";
+/// Versions for boogie, z3, and cvc5. The upgrade of boogie and z3 is mostly backward compatible,
+/// but not always. Setting the max version allows Prover to warn users for the higher version of
+/// boogie and z3 because those may be incompatible.
+const MIN_BOOGIE_VERSION: Option<&str> = Some("2.15.8");
+const MAX_BOOGIE_VERSION: Option<&str> = Some("2.15.8");
+
+const MIN_Z3_VERSION: Option<&str> = Some("4.11.0");
+const MAX_Z3_VERSION: Option<&str> = Some("4.11.0");
+
+const MIN_CVC5_VERSION: Option<&str> = Some("0.0.3");
+const MAX_CVC5_VERSION: Option<&str> = None;
+
+enum Op {
```

Comment:
As also mentioned by @brmataptos, to minimize the code change, you actually don't need this enum to distinguish two cases.

Reply:
I have removed enum.